RFC/src/RFC-0173_Versioning.md

```diff
@@ -0,0 +1,133 @@
+is speaking on, such as TestNet or Mainnet.
+
+Tari message packets are encapsulated using the Noise protocol so we do not need the delimiting functionality of these 
+bytes but Tari will include a single WireMode byte at the beginning of every connection session. This byte will indicate 
+which network a node is communicating on, so that if the counterparty is on a different network it can reject this 
+connection cheaply without having to perform any further operations, like completing the Noise protocol handshake.
+
+The following is a proposed mapping of the WireMode byte. Space is left between Mainnet, Stagenet and Testnet bytes for
+future use. 
+```rust,ignore
+   #[repr(u8)]
+   enum Network {
+      Mainnet = 0x00,
+      Stagenet1 = 0x51,
+      Testnet1 = 0xa1,
+      Testnet2 = 0xa2
+   }
+```
+### P2P message version
+Peer to Peer messages on the Tari network are encapsulated into message envelopes. The body of message envelopes are 
+defined, serialized and deserialized using Protobuf. These messages will only be updated by adding new fields to the 
+Protobuf definitions, never removing fields. This is done in order to preserve backwards compatibility where newer nodes 
+can still communicate with older nodes. 
+
+The P2P messaging protocol will see many changes in its lifetime. Some will be minor changes that are fully backwards 
+compatible and some changes will be breaking where older nodes will not be able to communicate with newer nodes. In 
+order to document these two classes of changes each P2P message header will contain a `version` field that will use
+a two-part semantic versioning scheme with the format of `major.minor` integer versions. The `minor` version will be 
+incremented whenever there is any change. The `major` version be incremented when there is a breaking change made to 
+the P2P protocol. Each integer can be stored separately.
+
+### Consensus version
+The final aspect of the Tari system that will be versioned are the Consensus rules. These rules will change as the 
+network matures. Changes to consensus rules can be achieved using either a Soft fork or Hard fork. Soft forks are where 
+new consensus rules are added that older nodes will see as valid (thus backwards compatible) but newer nodes will reject 
+blocks from older nodes that are not aware of the new consensus rules. A hard fork means that the new consensus rules 
+are not backwards compatible and so only nodes that know about the new rules will be able to produce and validate new 
+transactions and blocks.
+
+The consensus version will be used by a node to determine if it can interact with another node successfully or not. A 
+list of fork versions will be maintained within the code. When a connection is started with a new node the two nodes 
+will exchange `Version` messages detailing the consensus version they are each running and the blockheight at which they
```

Comment:
The Tari protocol already has provision for this - we should probably describe that here.